### ISSUE TITLE
Setup auto-generated API docs

### ISSUE THREAD

**Issue by jonas-eschle** [Phynix/uniphy#15]
In order to have always the newest API docs, an autogeneration is the way to go. Open questions are:

- Where to put it? (github pages seems good as I think it closely belongs to the project anyway)
- Style? An often used one which I would propose to use the "sphinx_rtd_theme" as used here: https://mayou36.bitbucket.io/raredecay/docs/
- to plan for long terms (at least after the first 1.0 release) I think it is crucial to have all versions documentation available and not delete old one.

Anything else @SebastianJL ?

**Comment by SebastianJL** [comment]
I agree with all of your points. Some further considerations:
- Where will the docs be saved? One could use a /docs folder or an entire gh-pages branch. I think I tend to the latter as it separates the code from the doc just enough to keep readability while not pushing it too far away. Of course this kind of defeats the purpose of git branches as the contents of this branch will be entirely different from the master and are never intended to be merged. There would also be the possibility to create an entirely new project just for the docs.
- Do you already know how to generate the docs? There's an example [here](https://daler.github.io/sphinxdoc-test/includeme.html).
- I think the versioning of the docs can be implemented later but before the release of version 1.0.0.

**Comment by jonas-eschle** [comment]
Good point! I tried to look around how others do it and whats may better, after that I would probably prefer the /docs, because of the following:
  - gh-pages seems to be the "old" way, "/docs" on master the "new" way
  - although they are slightly closer to the code with this approach, I think this is fine as they will be on the "same level" as models, tests etc. (which seems appropriate).
  - it could be easier to handle as no extra effort (changing branches, versioning etc.) is needed. The docs are in docs and are always on the same status as the whole project.
On the other hand, I don't really see a benefit in using gh-pages, or is there one?

Yes, I do, I will even use the autodoc utility.

Agree.

**Comment by SebastianJL** [comment]
Ok let's do it your way then. How long do you think this will take?

**Comment by jonas-eschle** [comment]
I am currently still busy with another project, but I think in one week this should be done.
